src/randoop/NoExceptionCheck.java

```diff
@@ -3,8 +3,10 @@
 import randoop.sequence.Execution;
 
 /**
- * Checks that no exception is thrown after a given statement
- * in a sequence.
+ * Checks that an exception is not thrown by a given statement in a sequence. 
+ * This is meant for an error-revealing check where, in fact, an exception
+ * was observed at the statement, and this check is meant to represent that
+ * the throws is actually an expected failure.
```

Comment:
"expected failure" seems strange.  Also, "throws" sounds like a static code structure; do you mean the thrown exception?

After reading this a few times, I think it means:
 * The code under test is not supposed to throw an exception.  
 * The generated test checks for an exception and fails if one is thrown.
 * (Implementation detail that should be stated at the end:  This kind of test is generated when an exception was observed during test generation.)

I think it's most helpful to first state what something is or means, and then discuss typical uses.  Those uses can be illuminating, but mixing them together with the specification proper is confusing.